script.js

```diff
@@ -0,0 +1,41 @@
+var tester = function() {
+    console.log('Button Pushed');
+    var fileHeight = document.getElementById('heightNumber').value;
+    console.log(fileHeight);
+    var madeOf = document.getElementById('charType').value;
+    console.log(madeOf);
+        (function () {
```

Comment:
Is there a reason you chose to make this an IIFE? Doing without works just fine.

```js
var tester = function() {
    console.log('Button Pushed');
    var fileHeight = document.getElementById('heightNumber').value;
    console.log(fileHeight);
    var madeOf = document.getElementById('charType').value;
    console.log(madeOf);

     var spc = [];
     var leafs = [];
     var finaltree = []

     for (i = fileHeight; i >=0; i--) {
         spc.push(' '.repeat(i));
     }
 //console.log(spc);

     for (j = 1; j < (fileHeight * 2); j = j + 2) {
         leafs.push(madeOf.repeat(j));
    }
 //console.log(leafs);
     for (z = 0; z < fileHeight; z++) {
         finaltree.push(spc[z], leafs[z]);
         console.log(finaltree.join(' '));
         finaltree = [];
    }
}
```